### DISCUSSION TITLE
ErrorLog - RabbitMQ-SendEmailQueue,Already closed: The AMQP operation was interrupted: AMQP close-reason, initiated by Library, code=541, text='Unexpected Exception', classId=0, methodId=0, cause=System.Collections.Generic.KeyNotFoundException: 给定关键字不在字典中。

### DISCUSSION THREAD

**Discussion by mq00fc** [rabbitmq/rabbitmq-dotnet-client#1400] [General]
### Describe the bug

ErrorLog - RabbitMQ-SendEmailQueue,Already closed: The AMQP operation was interrupted: AMQP close-reason, initiated by Library, code=541, text='Unexpected Exception', classId=0, methodId=0, cause=System.Collections.Generic.KeyNotFoundException: 给定关键字不在字典中。

### Reproduction steps

idk..

### Expected behavior

This problem occurs when using the right-click to send the queue, but it is not clear why this occurs.

### Additional context

ErrorLog - RabbitMQ-publishMessageToServer:GaoXinCashShop_SendEmailQueue,Already closed: The AMQP operation was interrupted: AMQP close-reason, initiated by Library, code=541, text='Unexpected Exception', classId=0, methodId=0, cause=System.Collections.Generic.KeyNotFoundException: 给定关键字不在字典中。
   在 System.ThrowHelper.ThrowKeyNotFoundException()
   在 System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   在 RabbitMQ.Client.Impl.SessionManager.Lookup(Int32 number)
   在 RabbitMQ.Client.Framing.Impl.Connection.MainLoopIteration()
   在 RabbitMQ.Client.Framing.Impl.Connection.MainLoop()
{"Title":"产品上传通知","Content":"您于2023/10/9 17:11:34提交的产品上传现已完成\n导入384818行,成功:384818,完成时间为:2023/10/9 17:58:17,如下是错误信息\n\n","UserEmail":"","isHtml":false}
RabbitMQ.Client.Exceptions.AlreadyClosedException: Already closed: The AMQP operation was interrupted: AMQP close-reason, initiated by Library, code=541, text='Unexpected Exception', classId=0, methodId=0, cause=System.Collections.Generic.KeyNotFoundException: 给定关键字不在字典中。
   在 System.ThrowHelper.ThrowKeyNotFoundException()
   在 System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   在 RabbitMQ.Client.Impl.SessionManager.Lookup(Int32 number)
   在 RabbitMQ.Client.Framing.Impl.Connection.MainLoopIteration()
   在 RabbitMQ.Client.Framing.Impl.Connection.MainLoop()
   在 RabbitMQ.Client.Framing.Impl.Connection.EnsureIsOpen()
   在 RabbitMQ.Client.Framing.Impl.AutorecoveringConnection.CreateModel()
   在 GaoXinCoreCashShop.Common.RabbitMQClientHandler.publishMessageToServer[T](String queueName, T model) 位置 C:\Users\Administrator.DESKTOP-1A2DGNK\Desktop\GaoXinSourceCode\GaoXinCoreCashShop\GaoXinCoreCashShop.Core\Common\RabbitMQClientHandler.cs:行号 247

**Comment by michaelklishin** [comment]
No, the error does not occur when you right click something. The connection is closed earlier (as `RabbitMQ.Client.Exceptions.AlreadyClosedException` indicates), and without a log snippet from the server plus a piece of code that demonstrates what your application does, we would not be able to suggest anything more specific.

We do not guess in this community.

What is particularly confusing is that the exception seemingly originates in `RabbitMQ.Client.Impl.SessionManager.Lookup(Int32 number)` but the provided value is a string.

Please put together a small executable example that reproduces the problem and publish it on GitHub so that the maintainers of this client could take a traffic capture and troubleshoot further.
